rust/private/providers.bzl

```diff
@@ -49,6 +50,7 @@ DepInfo = provider(
         "transitive_build_infos": "depset[BuildInfo]",
         "transitive_crate_outputs": "depset[File]: All transitive crate outputs.",
         "transitive_crates": "depset[CrateInfo]",
+        "transitive_metadata_outputs": "depset[File]: All transitive crate metadata outputs.",
```

Comment:
I believe this description is not correct anymore, as we're now adding also `.rlib`s from the dependencies that don't have metadata enabled. Maybe something along the lines of `Transitive outputs for pipelined compilation` fits better?